src/libraries/System.Runtime/tests/TrimmingTests/InvariantGlobalizationTrue.cs

```diff
@@ -29,32 +27,37 @@ static int Main(string[] args)
         {
             return -3;
         }
+        
+        // Ensure the internal GlobalizationMode class is trimmed correctly.
+        Type globalizationMode = typeof(object).Assembly.GetType("System.Globalization.GlobalizationMode", throwOnError: false);
 
-        foreach (MemberInfo member in globalizationMode.GetMembers(allStatics))
+        if (OperatingSystem.IsWindows())
         {
-            // properties and their backing getter methods are OK
-            if (member is PropertyInfo || member.Name.StartsWith("get_"))
+            foreach (MemberInfo member in globalizationMode.GetMembers(allStatics))
             {
-                continue;
-            }
+                // properties and their backing getter methods are OK
+                if (member is PropertyInfo || member.Name.StartsWith("get_"))
+                {
+                    continue;
+                }
 
-            if (OperatingSystem.IsWindows())
-            {
-                // Windows still contains a static cctor and a backing field for UseNls
+                // Windows still contains a static cctor and a backing field for UseNls.
                 if (member is ConstructorInfo || (member is FieldInfo field && field.Name.Contains("UseNls")))
                 {
                     continue;
                 }
-            }
 
-            // Some unexpected member was left on GlobalizationMode, fail
-            Console.WriteLine($"Member '{member.Name}' was not trimmed from GlobalizationMode, but should have been.");
+                // Some unexpected member was left on GlobalizationMode, fail
+                Console.WriteLine($"Member '{member.Name}' was not trimmed from GlobalizationMode, but should have been.");
+                return -4;
+            }
+        }
+        // On non Windows platforms, the full type is trimmed.
+        else if (globalizationMode is not null)
+        {
             return -4;
```

Comment:
could you please ass Console.WriteLine($"It is expected to have System.Globalization.GlobalizationMode type trimmed on Non-Windows platforms")